ibis/tests/all/test_generic.py

```diff
@@ -1,48 +1,28 @@
 import decimal
 
+import pandas as pd
 import pytest
 
 import ibis
 from ibis import literal as L
-from ibis.tests.backends import (
-    BigQuery,
-    Clickhouse,
-    MySQL,
-    OmniSciDB,
-    Postgres,
-    PySpark,
-    Spark,
-    SQLite,
-)
 
 
 @pytest.mark.parametrize(
     ('expr', 'expected'),
     [
         (ibis.NA.fillna(5), 5),
         (L(5).fillna(10), 5),
-        pytest.param(
-            L(5).nullif(5),
-            None,
-            marks=pytest.mark.xpass_backends(
-                [
-                    BigQuery,
-                    Clickhouse,
-                    MySQL,
-                    OmniSciDB,
-                    Postgres,
-                    Spark,
-                    SQLite,
-                    PySpark,
-                ]
-            ),
-        ),
+        (L(5).nullif(5), None),
         (L(10).nullif(5), 10),
     ],
 )
 @pytest.mark.xfail_unsupported
 def test_fillna_nullif(backend, con, expr, expected):
-    assert con.execute(expr) == expected
+    if expected is None:
+        # We are being flexible with how nulls are represented by each backend
```

Comment:
Not sure about. Is this a behavior difference specific to Spark 3?

Reply:
Yep I am doing this because PySpark 3 returns `np.nan` rather rather than `None`

However, the Csv, Pandas, and Parquet backends _also_ return `np.nan` so this change (including removing `marks=pytest.mark.xpass_backends(...)` above this line) lets us run this test on all backends rather than just the ones that return `None`

Reply:
Hmm. OK, if this is a difference among all backends then I think it should be addressed outside this PR. Can you create an issue for this?

Reply:
Ok will do

Are you OK with keeping this change in this PR anyways? If so I can note the issue as a comment in the code here. Otherwise, I can also just remove PySpark and Spark from being run on this test.

Reply:
I am OK with keep the change in this PR

Reply:
Sounds good, I created issue #2365 and now am mentioning it in a comment in the code here.